examples/canopen/can_disturbance.py

```diff
@@ -19,11 +20,44 @@ def disturbance_example():
     # A = signal_amplitude (Amplitude)
     # t = sample_period*i (time)
     # w = signal_frequency*2*math.pi (angular frequency)
-    data_pos = [int(1000*signal_amplitude * math.sin(sample_period*i * signal_frequency * 2*math.pi)) for i in range(n_samples)]
-    data_vel = [signal_amplitude * math.sin(sample_period*i * signal_frequency * 2*math.pi) for i in range(n_samples)]
-    data_curr_q = [signal_amplitude * math.sin(sample_period*i * signal_frequency * 2*math.pi - math.pi/2) for i in range(n_samples)]
-    data_curr_d = [signal_amplitude * math.sin(sample_period*i * signal_frequency * 2*math.pi + math.pi/2) for i in range(n_samples)]
-    data_positioning_opt = [int(abs(500*signal_amplitude * math.sin(sample_period*i * signal_frequency * 2*math.pi + math.pi))) for i in range(n_samples)]
+    data_pos = cast(
+        List[Union[int, float]],
```

Comment:
I don't understand why a explicit cast is necessary here

Reply:
Mypy will complain if we don't since having a list of numbers that can be int or float is essentially too uncertain for the type check to work.
I'm not sure I entirely understand how this works or why it's a problem, but the explicit cast makes mypy happy enough.
More info:
https://stackoverflow.com/questions/43957034/specifying-a-type-to-be-a-list-of-numbers-ints-and-or-floats

Reply:
Oh, I see the problem.
The method argument disturbance_write_data is not correctly typed.
It accepts a int/float covariant list, and since the list is not immutable the variance could be change by the method and force a new internal type.
I think we should review this but maybe not in this issue. Forcing the user to use so many verbose casts seems ridiculous to me